src/utils/DAGNode.ts

```diff
@@ -0,0 +1,29 @@
+import * as Store from 'orbit-db-store';
+
+export class DAGNode {
+    nodeList: [DAGNode];
+    hash: string;
+
+    constructor(hash, nodeList) {
+        this.nodeList = nodeList;
+        this.hash = hash;
+    }
+
+    static emptyDAG(): DAGNode {
+        return new DAGNode("EMPTY", []);
+    }
+
+    static createDAG(head: any, db: Store, depth: number = 10): DAGNode {
+        return depth > 0
+            ? new DAGNode(
+                head.hash,
+                (head.next
+                    ? head.next.map(node => this.createDAG(db.get(node), db, --depth))
```

Comment:
Dear Matthew,

The next list does not represent the children of the current node, but rather the entire list of logs from the current node to the beginning, so the child should just be the first element of the next list.